### PR TITLE
Add TensorFlow backend

### PR DESCRIPTION
**Pull request by Thommy257**
This PR adds a TensorFlow backend to discopy. This is required to support hybrid quantum-classical models in lambeq.

### CODE REVIEW

**Comment by y-richie-y** [comment]
We want to be able to backpropagate using tensorflow, so we shouldn't use `tf.constant`

File: discopy/tensor.py [hunk 1]
```diff
@@ -103,12 +103,22 @@ def __init__(self):
         self.array = torch.as_tensor
 
 
+class TensorFlowBackend(TensorBackend):
+    def __init__(self):
+        import tensorflow as tf
+        from tensorflow.python.ops.numpy_ops import np_config
+        np_config.enable_numpy_behavior()
+        self.module = tf
+        self.array = tf.constant
```
Comment (Thommy257) [inline discopy/tensor.py:112]:
```suggestion
        self.array = tf.Variable
```

**Comment by Thommy257** [comment]
Is there anything I should add to this PR then?

**Comment by Thommy257** [comment]
There's the double/float error again (using pytorch). Could you have a look at the notebook at some point @y-richie-y? Not urgent though!

File: discopy/tensor.py [hunk 2]
```diff
@@ -214,8 +223,12 @@ def __complex__(self):
         return complex(self.array)
 
     def __repr__(self):
+        if hasattr(self.array, 'numpy'):
+            np_array = self.array.numpy()
+        else:
+            np_array = self.array
         return "Tensor(dom={}, cod={}, array={})".format(
-            self.dom, self.cod, array2string(self.array.flatten()))
+            self.dom, self.cod, array2string(np_array.reshape(-1)))
```
Comment (Thommy257) [inline discopy/tensor.py:231]: `tensorflow.numpy` does not have a `.flatten()` mehtod.

**Pull request merged**